apps/snitch_api/lib/snitch_api_web/controllers/product_controller.ex

```diff
@@ -0,0 +1,78 @@
+defmodule SnitchApiWeb.ProductController do
+  use SnitchApiWeb, :controller
+
+  alias SnitchApi.API, as: Context
+
+  action_fallback(SnitchApiWeb.FallbackController)
+
+  # sort the products in Z-A descending order_by prodcut name
+  def index(conn, %{"sort" => "Z-A"} = params) do
+    {products, page} = Context.products_by_desc(params, conn)
```

Comment:
Have you tested error case for the actions, like are you getting fallback responses on error?